### PR TITLE
Allow OIDC auth in debug TestAuth mode

### LINKED ISSUE #399: Allow Debug auth to use interactive OIDC while preserving TestAuth bootstrap
Objective:
- Make Grace Debug/TestAuth configuration accept normal interactive OIDC/MSA-authenticated CLI requests for authenticated endpoints such as `GET /auth/me` and `POST /auth/token/create`, while preserving the existing no-external-auth TestAuth bootstrap path for first-time local environments.

Context and evidence:
- Current Debug server logs show `GraceTest` challenged and 401 responses for `POST /auth/token/create` and `GET /auth/me` even when an Authorization header is present.
- `src/Grace.Server/Security/TestAuth.Server.fs` authenticates only when `x-grace-user-id` is present.
- `src/Grace.Server/Security/AuthSchemeSelection.Server.fs` currently routes non-PAT bearer requests to `GraceTest` when `GRACE_TESTING=1`.
- `src/Grace.Server/Startup.Server.fs` configures the testing auth branch with `GraceTest` and `GracePat`, but does not make OIDC/JWT bearer auth available in the testing branch.
- `src/Grace.CLI/Command/Auth.CLI.fs` gets interactive tokens for SDK calls, but `GRACE_TOKEN` intentionally remains PAT-only.
- `scripts/start-debuglocal.ps1` already proves the first-time local bootstrap path with explicit `x-grace-user-id` and PAT creation. Keep that path working.
- Local planning artifact: `docs/DebugAuthLocalBootstrapPlan.html`.

Owned paths:
- `src/Grace.Server/Startup.Server.fs`
- `src/Grace.Server/Security/AuthSchemeSelection.Server.fs`
- `src/Grace.Server/Security/TestAuth.Server.fs` only if the explicit TestAuth contract needs a small additive helper or test seam
- `src/Grace.Server/Security/*.Server.fs` only for auth-scheme composition directly required by this issue
- `src/Grace.SDK/Common.SDK.fs`
- `src/Grace.SDK/Auth.SDK.fs`
- `src/Grace.SDK/PersonalAccessToken.SDK.fs`
- `src/Grace.CLI/Command/Auth.CLI.fs`
- `src/Grace.Authorization.Tests/**`
- `src/Grace.Server.Unit.Tests/**`
- `src/Grace.Server.Tests/**` if HTTP-pipeline proof is needed
- `src/Grace.CLI.Tests/**`
- `src/docs/ENVIRONMENT.md`
- `docs/**` for first-time stand-up and Debug auth guidance
- nearby `AGENTS.md` files only if future agents need updated auth workflow context

Forbidden or sensitive paths:
- Do not re-enable local token-file persistence or broaden `GRACE_TOKEN` beyond Grace PATs.
- Do not make `/auth/token/create` anonymous or bypass authorization/role checks for token creation.
- Do not teach `GraceTest` to trust arbitrary bearer tokens.
- Do not log bearer tokens, PATs, authorization headers, secrets, or credential-bearing URLs.
- Do not change unrelated owner/repository command behavior, including the separate OwnerId issue, without updating the issue first.
- Do not edit Aspire resource, storage, Service Bus, Redis, or deployment infrastructure unless live evidence proves it is required and the issue is updated before editing.

Risk surfaces:
- Auth or secrets
- CLI public contract
- Server or API contract
- SDK or client contract
- Docs or workflow

Minimum detail gate:
- Invariant tuple:
  - Actor: Grace CLI user running against Debug/`GRACE_TESTING=1` server.
  - Routes: `GET /auth/me` and `POST /auth/token/create` remain authenticated endpoints.
  - Schemes: Grace PAT bearer uses `GracePat`; valid non-PAT OIDC/JWT bearer uses JwtBearer when OIDC config exists; explicit `x-grace-user-id` uses `GraceTest` only in testing mode.
  - Identity dimensions: the effective Grace user identity comes from the validated principal claim mapping for OIDC/PAT, or from the explicit TestAuth header for local/test bootstrap; mixed or stale identities must not silently collapse into the wrong user.
  - Durable source of truth: server authentication/authorization middleware and PAT issuance state, with RBAC/role assignment still authorizing token creation after authentication succeeds.
- Forbidden implementation shapes:
  - No accepting unvalidated bearer tokens in Debug.
  - No using TestAuth as a fallback for arbitrary Authorization headers.
  - No stripping Authorization headers client-side to force TestAuth.
  - No changing `GRACE_TOKEN` to accept Auth0/OIDC access tokens.
  - No broad catch-all exception handling that converts auth failures into success or hides server errors.
  - No docs that tell maintainers to manually mint tokens with raw `Invoke-RestMethod` as the normal path.
- Expected tests:
  - positive: PAT bearer still selects `GracePat` and authenticates.
  - positive: explicit `x-grace-user-id` still authenticates in Debug/TestAuth mode.
  - positive: valid non-PAT OIDC/JWT bearer is routed to JwtBearer in Debug/TestAuth mode when OIDC config exists.
  - positive: authenticated OIDC principal can call `/auth/token/create` when authorized.
  - negative: absent auth, invalid bearer, or OIDC bearer without OIDC config does not authenticate.
  - negative: TestAuth is not selected merely because a non-PAT Authorization header is present.
  - regression: `grace auth whoami -o Verbose` and token creation no longer surface empty-body 401 JSON parse exceptions.
  - boundary: PAT prefix checks remain exact and do not misclassify malformed, blank, or whitespace bearer values.
- High-risk adversarial examples:
  - Request contains both `Authorization: Bearer <Grace PAT>` and `x-grace-user-id`; PAT precedence must be explicit and tested or deliberately rejected.
  - Request contains non-PAT bearer plus `x-grace-user-id`; selector behavior must be deterministic and documented/test-covered.
  - `GRACE_TESTING=1` with no OIDC config must not attempt JwtBearer validation or produce misleading CLI guidance.
  - `GRACE_TESTING=0` must preserve production OIDC/PAT behavior.
  - Empty 401 response bodies must render actionable client errors without leaking token material.
- Selected risk-surface traps:
  - Auth ordering: prove scheme selection order, not only happy-path endpoint execution.
  - CLI side effects: preserve stdout/stderr cleanliness and `--output Json` contract where changed.
  - Contract propagation: update SDK/CLI error handling together if the error envelope behavior changes.
  - Docs freshness: document both first-time local TestAuth bootstrap and interactive OIDC/MSA bootstrap; call out first-admin authorization separately from authentication.
  - Security logging: keep Authorization/PAT material redacted in logs and tests.
- Explicit N/A waivers with reasons:
  - Orleans actor behavior: N/A unless implementation changes PAT issuance grain behavior; expected change is auth composition and client error handling.
  - Storage/Cosmos/Service Bus/Redis: N/A unless worker discovers token persistence semantics require integration proof beyond existing tests.
  - OpenAPI route shape: N/A if route paths, methods, request DTOs, and response DTOs do not change.

Validation:
- Focused command:
  - Worker should first add focused failing tests around auth scheme selection and client 401 handling, then run the smallest matching focused test commands needed for RED/GREEN evidence.
  - If using `dotnet test --no-build`, first build the matching project in Release.
- Formatting:
  - Run targeted Fantomas formatting/checks before build/test validation for touched F# files.
- Fast repo gate:
  - `pwsh ./scripts/validate.ps1 -Fast`
- Full or Aspire gate, if needed:
  - `pwsh ./scripts/validate.ps1 -Full` only if the worker changes Aspire/runtime startup behavior or cannot prove the HTTP auth pipeline through Fast-profile tests.
- Docs validation:
  - `npx --yes markdownlint-cli2 "src/docs/ENVIRONMENT.md"`
  - Include any other changed Markdown docs in the same MarkdownLint command.
- Final hygiene:
  - `git diff --check`
- Manual verification, if practical:
  - In Debug with OIDC configured: `grace auth login`, `grace auth whoami -o Verbose`, then `grace auth token create --token-name local-dev --expires-in 90d -o Verbose`.
  - In Debug without OIDC setup: `pwsh ./scripts/start-debuglocal.ps1 -GraceServerUri "http://localhost:5000"` still prints usable `GRACE_SERVER_URI` and `GRACE_TOKEN` values.

Definition of done:
- Debug/TestAuth mode is additive: PAT, OIDC/JWT bearer, and explicit TestAuth each route to the intended scheme.
- `grace auth token create` works after interactive OIDC/MSA login when the authenticated user is authorized.
- First-time stand-up remains easy through `start-debuglocal.ps1` with explicit TestAuth bootstrap.
- Empty-body 401s from auth endpoints no longer produce raw JSON parse exceptions in CLI/SDK-facing auth commands.
- `/docs` guidance is updated for Debug auth, first-time local bootstrap, interactive OIDC/MSA bootstrap, and first-admin authorization.
- Tests cover positive, negative, regression, and boundary cases from this issue.
- Coding and fix work completed through implementation subagents, with the main agent acting as orchestrator.
- Worker completed a bot-prevention self-review over the actual diff before handoff.
- Ready-for-review pull request opened and linked with `Closes #<issue>`.
- Validation evidence recorded on the PR.
- Codex Code Review Bot reviewed the latest PR commit and either reported no issues with a thumbs-up reaction on the PR body or all bot comments were fixed, answered, resolved, and followed by a no-issues bot review.
- Follow-ups named explicitly if any behavior is intentionally deferred.

### PR DESCRIPTION
**Pull request by ScottArbeit**
## Summary

- Allows Debug/TestAuth mode to use normal OIDC/JWT bearer authentication when OIDC configuration exists, while preserving Grace PAT and explicit TestAuth paths.
- Improves auth client handling for empty-body 401 responses so CLI auth commands surface actionable errors instead of JSON parse exceptions.
- Updates Debug/local auth documentation for first-time TestAuth bootstrap and interactive OIDC/MSA bootstrap.

Closes #399

## Validation

- `dotnet tool restore`: passed after NuGet access; restored repo-pinned `fantomas-tool` `4.7.9`.
- `dotnet C:\Users\scott\.nuget\packages\fantomas-tool\4.7.9\tools\netcoreapp3.1\any\fantomas-tool.dll <touched F# files>`: passed.
- `dotnet build --configuration Release src/Grace.Server.Unit.Tests/Grace.Server.Unit.Tests.fsproj`: passed.
- `dotnet test --no-build --configuration Release src/Grace.Server.Unit.Tests/Grace.Server.Unit.Tests.fsproj --filter "FullyQualifiedName~AuthSchemeSelection"`: passed, 13 tests.
- `dotnet build --configuration Release src/Grace.CLI.Tests/Grace.CLI.Tests.fsproj`: passed.
- `dotnet test --no-build --configuration Release src/Grace.CLI.Tests/Grace.CLI.Tests.fsproj --filter "Name~empty"`: passed, 5 tests.
- `npx --yes markdownlint-cli2 "src/docs/ENVIRONMENT.md" "docs/Authentication.md" "src/Grace.Authorization.Tests/AGENTS.md"`: passed, 0 errors.
- `pwsh ./scripts/validate.ps1 -Fast`: passed. Fast test set reported Authorization 39, Types 96, Server.Unit 503, CLI 510; `Grace.Server.Tests` had no matches for the Fast filter.
- `git diff --check`: passed.
- GitHub Actions `full`: passed on `13b53cc9d233803977fce8f3d4edf6fc385821cc` in 5m45s.

Review fix validation:

- `npx --yes markdownlint-cli2 "AGENTS.md" "docs/Authentication.md" "docs/Development process.md" "src/AGENTS.md" "src/docs/ENVIRONMENT.md"`: passed, 0 errors.
- Focused/script/validate command: skipped because the review fix is docs-only; no script or F# behavior changed.
- `git diff --check`: passed.

## Review Status

- Implementation worker: completed in `C:\Source\Grace-gh-399` on `agent/399-debug-auth-oidc-testauth`.
- Worker bot-prevention self-review: completed over auth scheme precedence, mixed Authorization/TestAuth headers, missing OIDC config, PAT precedence, anonymous-token-creation avoidance, client-side header behavior, secret redaction, docs clarity, and formatter churn.
- Codex Code Review Bot: reviewed `fa0f616f64238547c65f4a1b09c21c498cbada5c` and opened one P2 finding in `docs/Authentication.md` line 142.
- Fix commit: `13b53cc9d233803977fce8f3d4edf6fc385821cc` updated zero-OIDC bootstrap docs to use `-SkipAuthProbe` and added the 120-second maximum sleep/poll command rule to agent docs.
- Bot finding response: replied to `discussion_r3410547567` with fix and validation evidence; conversation resolved.
- Final Codex Code Review Bot state: no-issues thumbs-up reaction observed on PR body for current head `13b53cc9d233803977fce8f3d4edf6fc385821cc`; no unresolved bot threads remain.

## Docs Impact

- Updated Debug/local auth guidance in `/docs`, including `src/docs/ENVIRONMENT.md` and `docs/Authentication.md`.
- Added project-local authorization-test guidance in `src/Grace.Authorization.Tests/AGENTS.md`.
- Updated `AGENTS.md`, `src/AGENTS.md`, and `docs/Development process.md` to state that agents must not sleep or poll for more than 120 seconds in one command.

## Residual Risk

- A live external OIDC/MSA token smoke was not run locally. Coverage is deterministic scheme selection, testing startup registration for JwtBearer when OIDC config exists, PAT/TestAuth precedence tests, and CLI empty-body 401 regression tests.
- Grace pins `fantomas-tool` `4.7.9`; avoid the user-global Fantomas `7.0.3` for this branch to prevent unrelated formatter churn.

### CODE REVIEW

File: docs/Authentication.md [hunk 1]
```diff
@@ -126,8 +136,46 @@ When enabled, Grace authenticates requests using headers:
    curl -H "x-grace-user-id: dev-scott" "http://localhost:5000/auth/me"
    ```
 
-On first activation (with no existing assignments), Grace will seed `SystemAdmin` for `dev-scott`.
-After that, bootstrap is a no-op.
+On first activation (with no existing assignments), Grace will seed `SystemAdmin` for `dev-scott`.
+After that, bootstrap is a no-op.
+
+The canonical zero-external-auth bootstrap helper uses this path and prints a usable PAT:
```
[the remark this answers is not in the document]
Comment (ScottArbeit) [reply docs/Authentication.md:142]: ## Review/Fix: Zero-OIDC bootstrap probe

**Status:** Fixed in `13b53cc9d233803977fce8f3d4edf6fc385821cc`
**Review source:** Codex Code Review Bot
**Validation:** `npx --yes markdownlint-cli2 "AGENTS.md" "docs/Authentication.md" "docs/Development process.md" "src/AGENTS.md" "src/docs/ENVIRONMENT.md"` passed; `git diff --check` passed
**Prevention:** validation / stale-evidence gap; current issue and agent docs updated

### Summary

The zero-external-auth docs now show `-SkipAuthProbe` for the true no-OIDC bootstrap path, and explain why the switch is needed before TestAuth PAT creation. The same fix also records the new agent rule that sleep/poll commands must stay at or below 120 seconds.

### Review Issue

The documented no-OIDC path could abort before PAT creation because the default startup auth probe checks `/auth/oidc/config` first when OIDC settings are absent.

### Fix

Updated `docs/Authentication.md` and `src/docs/ENVIRONMENT.md` to use and explain `-SkipAuthProbe` for true zero-OIDC bootstrap. Updated `AGENTS.md`, `src/AGENTS.md`, and `docs/Development process.md` with the 120-second maximum sleep/poll command rule.

### Validation

- `npx --yes markdownlint-cli2 "AGENTS.md" "docs/Authentication.md" "docs/Development process.md" "src/AGENTS.md" "src/docs/ENVIRONMENT.md"`: passed, 0 errors
- Focused/script/validate command: skipped because the fix is docs-only; no script or F# behavior changed
- `git diff --check`: passed
